backend/app/tasks.py

```
@@ -3013,3 +3012,701 @@ def process_webapp(self, instance_alias, item_id, operation):
+
+    try:
+        progress_recorder.set_progress(0, 100, "Connecting to portal...")
+        portal_instance, target = _connect_portal(portal_alias, tool_result)
+        if not target:
+            progress_recorder.set_progress(100, 100, "Connection failed")
+        else:
+            if item_id:
+                # Webhook mode
+                progress_recorder.set_progress(20, 100, f"Processing item: {item_id}")
+                logger.info(f"[{portal_alias}] Public Unshare (webhook) for item {item_id}")
+                _handle_webhook_item(target, item_id, portal_instance, tool_result)
```

Ruff (0.12.2)

Undefined name `_handle_webhook_item`

(F821)